### PR TITLE
[meekrosoft-fff] Add new port

### PR DESCRIPTION
**Pull request by salman-javed-nz**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by salman-javed-nz** [comment]
@microsoft-github-policy-service agree

**Comment by salman-javed-nz** [comment]
Regarding PR checklist item "_The name of the port matches an existing name for this component on https://repology.org_", the name `fff` is also the name of a [file manager](https://github.com/dylanaraps/fff) written in Bash script.

I am not sure whether this is a genuine name clash, because I don't imagine the file manager fff is something one would want to pull with vcpkg. The most popular C project named `fff` on GitHub, going by the number of stars and forks, is the library in this PR.

File: ports/fff/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Header-only library
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO meekrosoft/fff
+    REF 7e09f07e5b262b1cc826189dc5057379e40ce886
```
Comment (FrankXie05) [inline ports/fff/portfile.cmake:6]:
```suggestion
    REF "v${VERSION}"
```
We can use the latest function.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Review by FrankXie05 — APPROVED** [review]

File: ports/fff/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Header-only library
```
Comment (JavierMatosD) [inline ports/fff/portfile.cmake:1]:
```suggestion
set(VCPKG_BUILD_TYPE release) # Header-only library
```

**Review by JavierMatosD — CHANGES_REQUESTED** [review]
Hi @salman-javed-nz, thank you for the contribution!

Do you mind changing the name of this port to `meekrosoft-fff` to disambiguate from other packages with the same name? I see there are multiple hits for `fff` on repology

**Comment by salman-javed-nz** [comment]
Both review comments addressed.

**Review by FrankXie05 — APPROVED** [review]

**Review by JavierMatosD — APPROVED** [review]

**Pull request merged**